draft-packages/pagination/KaizenDraft/Pagination.tsx

```diff
@@ -0,0 +1,168 @@
+import React from "react"
+import arrowBackward from "@kaizen/component-library/icons/arrow-backward.icon.svg"
+import arrowForward from "@kaizen/component-library/icons/arrow-forward.icon.svg"
+import { Icon } from "@kaizen/component-library"
+import cx from "classnames"
+import PageIndicator from "./PageIndicator"
+import TruncateIndicator from "./TruncateIndicator"
+import styles from "./styles.scss"
+
+export const createRange = (start: number, end: number): number[] => {
+  const range: number[] = []
+  for (let i = start; i <= end; i++) {
+    range.push(i)
+  }
+  return range
+}
+
+type Props = {
+  currentPage: number
+  pageCount: number
+  ariaLabelNextPage?: string
+  ariaLabelPreviousPage?: string
+  ariaLabelPage?: string
+  onPageChange: (newPage: number) => void
+}
+
+const boundaryPagesRange = 1
+const siblingPagesRange = 1
+
+const Pagination = ({
+  currentPage = 1,
+  pageCount,
+  ariaLabelNextPage,
+  ariaLabelPreviousPage,
+  ariaLabelPage,
+  onPageChange,
+}: Props) => {
+  // Click event for all pagination buttons (next, prev, and the actual numbers)
+  const handleButtonClick = (newPage: number | "prev" | "next") => {
+    if (newPage === "prev") {
+      onPageChange(currentPage - 1)
+      return
+    }
+    if (newPage === "next") {
+      onPageChange(currentPage + 1)
+      return
+    }
+    onPageChange(newPage)
+  }
+
+  const paginationIndicator = (page: number) => (
+    <PageIndicator
+      page={page}
+      selected={currentPage === page}
+      ariaLabelPage={ariaLabelPage}
+      onPageClick={handleButtonClick}
+    />
+  )
+
+  const truncateIndicator = (page: number) => <TruncateIndicator key={page} />
+
+  const pagination = () => {
+    const items: JSX.Element[] = []
+
+    // truncateSize is 1 now but could be 0 if we add the ability to hide it.
+    const truncateSize = 1
+
+    // Simplify generation of pages if number of available items is equal or greater than total pages to show
+    if (
+      1 + 2 * truncateSize + 2 * siblingPagesRange + 2 * boundaryPagesRange >=
```

Comment:
Feels like all the calculations happening here belong in a separate file with tests.